src/main/java/com/aol/cyclops/types/MonadicValue2.java

```diff
@@ -62,4 +66,25 @@
      * @see com.aol.cyclops.types.MonadicValue#unit(java.lang.Object)
      */
     <T2> MonadicValue2<T1, T2> unit(T2 unit);
+    
+	default <R> MonadicValue2<T1, R> flatMapIterable(Function<? super T2, ? extends Iterable<? extends R>> mapper) {
+		return this.flatMap(a -> {
+			Iterator<? extends R> it = mapper.apply(a).iterator();
+			if (it.hasNext()) {
+				R r = it.next();
+				return unit(r);
+			} else {
+				return null;
+			}
+		});
+	}
+
+	default <R> MonadicValue2<T1, R> flatMapPublisher(Function<? super T2, ? extends Publisher<? extends R>> mapper) {
```

Comment:
Same commend as for Iterable 

Could we get more detailed Javadoc here that explains we only take the first entry if there is more than 1 present?

(Then all the sub-classes can reference this definition too).

Thanks!